ignite/contrib/handlers/tensorboard_logger.py

```diff
@@ -419,20 +426,105 @@ class WeightsHistHandler(BaseWeightsHistHandler):
+
+            # Create a logger
+            tb_logger = TensorboardLogger(log_dir="experiments/tb_logs")
+
+            # Log weights which their name include 'conv'.
+            weight_selector = lambda name, p: 'conv' in name
+
+            # Attach the logger to the trainer to log weights norm after each iteration
+            tb_logger.attach(
+                trainer,
+                event_name=Events.ITERATION_COMPLETED,
+                log_handler=WeightsHistHandler(model, whitelist=weight_selector)
+            )
     """
 
-    def __init__(self, model: nn.Module, tag: Optional[str] = None):
+    def __init__(
+        self,
+        model: nn.Module,
+        tag: Optional[str] = None,
+        whitelist: Optional[Union[List[Union[str, nn.Module]], Callable[[str, nn.Parameter], bool]]] = None,
+    ):
         super(WeightsHistHandler, self).__init__(model, tag=tag)
 
+        self.weights = None
+        if whitelist is None:
+            return
+
+        self.weights = {}
+        if callable(whitelist):
+            for n, p in model.named_parameters():
+                if whitelist(n, p):
+                    self.weights[n] = p
+
+            if len(self.weights) == 0:
+                warnings.warn("Given callable whitelist does not " "select any parameter to be logged.")
+            return
+
+        for item in whitelist:
+            if isinstance(item, str):
+                try:
+                    for n, p in model.get_submodule(item).named_parameters(prefix=item):
+                        self.weights[n] = p
+                except AttributeError:
+                    try:
+                        self.weights[item] = model.get_parameter(item)
+                    except AttributeError:
+                        try:
+                            _ = model.get_buffer(item)
+                            raise ValueError(
+                                "Whitelist weights should not be Buffers as they are"
+                                f" considered persistent. Given buffer name `{item}`"
+                            )
+                        except AttributeError as e:
+                            raise ValueError(
+                                f"Whitelist weight `{item}` is not among model's parameters or submodules"
+                            ) from e
+            elif isinstance(item, nn.Module):
+                name = None
+                for n, m in model.named_modules():
+                    if m == item:
+                        name = n
+                        break
+                if name is not None:
+                    for n, p in item.named_parameters(prefix=name):
+                        self.weights[n] = p
+                else:
+                    raise ValueError(f"Module {item} is not among model's submodules")
+            else:
+                raise ValueError(f"Whitelist weights shoud be string or nn.Module, given {type(item)}")
```

Comment:
This part of code looks very bulky with all these try/except clauses.
Can we figure out how to simplify a bit the code ?

Reply:
I changed it. How is it?

Reply:
Well, still looks too complicated.
Let's try to simplify it in the following way:
```python
weights = {}
for name, param in model.named_parameters():    
    for item in whitelist:        
        if name.startswith(item):
            weights[name] = param
```
Here I assume that `item` is string. However, in your API `item` can be a `nn.Module`. The idea is to rewrite `whitelist` into `List[str]` and replace entries like `model.fc` or `model.layer3` by string values...

Reply:
You mean to remove entry type of callable as well?

Reply:
I changed `whitelist` to be `List[str]`. Is it OK now?

Reply:
Very sorry for delay @sadra-barikbin , actually, `callable` type was OK and gives a lot of flexibility.